client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -471,11 +472,11 @@ private boolean isFlowableByteArray(TypeToken entityTypeToken) {
     private Maybe<?> handleBodyReturnTypeAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
         final TypeToken entityTypeToken = TypeToken.of(entityType);
         final int responseStatusCode = response.statusCode();
-        final String httpMethod = methodParser.httpMethod();
+        final HttpMethod httpMethod = methodParser.httpMethod();
         final Type returnValueWireType = methodParser.returnValueWireType();
 
         final Maybe<?> asyncResult;
-        if (httpMethod.equalsIgnoreCase("HEAD")
+        if (httpMethod == HttpMethod.HEAD
```

Comment:
So does ExpandableStringEnum use an interned instance when `.fromString()` is called?

Reply:
Yeah. It checks to see if an instance already exists for the provided String value and will return it if so.